packages/nimble-blazor/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js

```diff
@@ -7,133 +7,145 @@
+                };
+            }
+        });
+        // Used by NimbleBanner.razor
+        Blazor.registerCustomEventType('nimblebannertoggle', {
+            browserEventName: 'toggle',
+            createEventArgs: event => {
+                return {
+                    newState: event.detail.newState,
+                    oldState: event.detail.oldState
+                };
+            }
+        });
+        // Used by NimbleTable.razor
+        Blazor.registerCustomEventType('nimbleactionmenubeforetoggle', {
+            browserEventName: 'action-menu-beforetoggle',
+            createEventArgs: event => {
+                return {
+                    newState: event.detail.newState,
+                    oldState: event.detail.oldState,
+                    recordIds: event.detail.recordIds,
+                    columnId: event.detail.columnId
+                };
+            }
+        });
+        // Used by NimbleTable.razor
+        Blazor.registerCustomEventType('nimbleactionmenutoggle', {
+            browserEventName: 'action-menu-toggle',
+            createEventArgs: event => {
+                return {
+                    newState: event.detail.newState,
+                    oldState: event.detail.oldState,
+                    recordIds: event.detail.recordIds,
+                    columnId: event.detail.columnId
+                };
+            }
+        });
+        // Used by NimbleTable.razor
+        Blazor.registerCustomEventType('nimbletablerowselectionchange', {
+            browserEventName: 'selection-change',
+            createEventArgs: event => {
+                return {
+                    selectedRecordIds: event.detail.selectedRecordIds
+                };
+            }
+        });
+    }
 }
 
-window.NimbleBlazor = {
-    Dialog: {
-        show: async function (dialogReference) {
-            const reason = await dialogReference.show();
-            return reason === window.customElements.get('nimble-dialog').UserDismissed;
-        },
-        close: function (dialogReference) {
-            dialogReference.close();
-        }
-    },
-    Drawer: {
-        show: async function (drawerReference) {
-            const reason = await drawerReference.show();
-            return reason === window.customElements.get('nimble-drawer').UserDismissed;
-        },
-        close: function (drawerReference) {
-            drawerReference.close();
-        }
-    },
-    Table: {
-        setData: async function (tableReference, data) {
-            const dataObject = JSON.parse(data);
-            await tableReference.setData(dataObject);
-        },
-        getSelectedRecordIds: async function (tableReference) {
-            return tableReference.getSelectedRecordIds();
-        },
-        setSelectedRecordIds: async function (tableReference, selectedRecordIds) {
-            await tableReference.setSelectedRecordIds(selectedRecordIds);
+window.NimbleBlazor = (window.NimbleBlazor === undefined)
```

Comment:
Is we choose to be more assertive then similar to above we can do a precondition check and throw instead of encapsulating all of this in a massive tertiary block